### ISSUE TITLE
pylsp doesn't start

### ISSUE THREAD

**Issue by rharish101** [dense-analysis/ale#4011]
## Information

**VIM version**

VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Dec 02 2021 19:44:33)
Included patches: 1-3582, 3602

Operating System: 
Arch Linux

## What went wrong

pylsp is detected, but ALE doesn't start it. ALE manages to start other linters like flake8, but not pylsp.

## Reproducing the bug

1. I installed pylsp through the Arch repos.
2. I setup ALE in my vimrc as follows:
    ```vim
    let g:ale_linters={
      \ 'python': ['pylsp'],
      \ 'tex': ['texlab', 'chktex'],
      \ }
    let g:ale_fixers={
      \ 'python': ['black', 'isort'],
      \ 'cpp': ['clang-format', 'clangtidy'],
      \ 'rust': ['rustfmt'],
      \ }
    let g:ale_python_pylsp_config={'pylsp': {
      \ 'configurationSources': ['flake8'],
      \ 'plugins': {
      \   'flake8': {'enabled': v:true},
      \   'pycodestyle': {'enabled': v:false},
      \   'pyflakes': {'enabled': v:false},
      \   'pydocstyle': {'enabled': v:true},
      \   'pylsp_mypy': {'dmypy': v:true},
      \ },
      \ }}
    let g:ale_completion_enabled=1 " enable ALE's completion through LSP
    set omnifunc=ale#completion#OmniFunc " make vim's omnicompletion use ALE, because supertab uses it for tab completion
    set completeopt=menu,menuone,noselect,noinsert " see `:h ale-completion-completeopt-bug`
    let g:ale_set_balloons=1 " enable ballon text using mouse hover through LSP
    ```
3. Next, I opened a python file inside a git repo with a `pyproject.toml` file.
4. ALE uses pylsp neither for linting nor for completion.

### :ALEInfo

```

 Current Filetype: python
Available Linters: ['bandit', 'cspell', 'flake8', 'flakehell', 'jedils', 'mypy', 'prospector', 'pycodestyle', 'pydocstyle', 'pyflakes', 'pylama', 'pylint', 'pylsp', 'pyre', 'pyright', 'vulture']
  Enabled Linters: ['pylsp']
  Ignored Linters: []
 Suggested Fixers: 
  'add_blank_lines_for_python_control_statements' - Add blank lines before control statements.
  'autoflake' - Fix flake issues with autoflake.
  'autoimport' - Fix import issues with autoimport.
  'autopep8' - Fix PEP8 issues with autopep8.
  'black' - Fix PEP8 issues with black.
  'isort' - Sort Python imports with isort.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'reorder-python-imports' - Sort Python imports with reorder-python-imports.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'yapf' - Fix Python files with yapf.
 Linter Variables:

let g:ale_python_auto_pipenv = 0
let g:ale_python_auto_poetry = 0
let g:ale_python_pylsp_auto_pipenv = 0
let g:ale_python_pylsp_auto_poetry = 0
let g:ale_python_pylsp_config = {'pylsp': {'plugins': {'pylsp_mypy': {'dmypy': v:true}, 'pydocstyle': {'enabled': v:true}}, 'configurationSources': ['flake8']}}
let g:ale_python_pylsp_executable = 'pylsp'
let g:ale_python_pylsp_options = ''
let g:ale_python_pylsp_use_global = 0
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = v:null
let g:ale_command_wrapper = ''
let g:ale_completion_delay = 100
let g:ale_completion_enabled = 1
let g:ale_completion_max_suggestions = 50
let g:ale_disable_lsp = 0
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {'rust': ['rustfmt'], 'cpp': ['clang-format', 'clangtidy'], 'python': ['black', 'isort']}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = v:null
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'tex': ['texlab', 'chktex'], 'python': ['pylsp']}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = v:null
let g:ale_list_window_size = v:null
let g:ale_loclist_msg_format = v:null
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = v:null
let g:ale_maximum_file_size = v:null
let g:ale_open_list = v:null
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_root = {}
let g:ale_set_balloons = 1
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = v:null
let g:ale_sign_error = v:null
let g:ale_sign_info = v:null
let g:ale_sign_offset = v:null
let g:ale_sign_style_error = v:null
let g:ale_sign_style_warning = v:null
let g:ale_sign_warning = v:null
let g:ale_sign_highlight_linenrs = v:null
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) pylsp
```

**Comment by hsanson** [comment]
Add this to your config:

```
call ch_logfile(expand('/tmp/chlogfile.log'), 'w')
```

Start vim and open a python file. Then inspect the /tmp/chlogfile.log file. It may contain some information on why pylsp is not starting.

**Comment by rharish101** [comment]
Thanks a lot @hsanson! I checked the log, and got to know that mypy wasn't properly installed. After fixing that, I got pylsp working again.

Anyway, shouldn't ALE have a command (maybe `:ALEDebug`?) or an option to `:ALEInfo` that prints the outputs, instead of manually dumping all logs using the above command?

**Comment by hsanson** [comment]
Indeed we want to add more LSP info to ALEInfo https://github.com/dense-analysis/ale/issues/3369.
